packages/lisk-transactions/test/utils/validation/validator.js

```diff
@@ -0,0 +1,1019 @@
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				timestamp: undefined,
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].params.missingProperty).to.equal('timestamp');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when timestamp is not integer', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				timestamp: 12.34,
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].dataPath).to.equal('.timestamp');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when senderId is not valid address', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				senderId: 'Invalid Address',
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].dataPath).to.equal('.senderId');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when senderPublicKey does not exist', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				senderPublicKey: undefined,
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].params.missingProperty).to.equal('senderPublicKey');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when senderPublicKey is not valid public key', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				senderPublicKey:
+					'a4465fd76c16fcc458448076372abf1912cc5b150663a64dffezzzzzzzzzzzzz',
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].dataPath).to.equal('.senderPublicKey');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when senderSecondPublicKey is not valid public key', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				senderSecondPublicKey:
+					'a4465fd76c16fcc458448076372abf1912cc5b150663a64dffezzzzzzzzzzzzz',
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].dataPath).to.equal('.senderSecondPublicKey');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when recipientId does not exist', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				recipientId: undefined,
+			});
+			const { valid, errors } = validateTransaction(invalidTransaction);
+			expect(errors[0].params.missingProperty).to.equal('recipientId');
+			return expect(valid).to.be.false;
+		});
+
+		it('should validate to be false with errors when recipientId is not valid address', () => {
+			const invalidTransaction = Object.assign({}, validTransaction, {
+				recipientId: '1234567891011121314151617181920L',
```

Comment:
Why is this invalid address different to the above example for `senderId`? Do we need two tests for each?

Reply:
it's tested in unit test for the function, but i think it's better to test different value than the same one